### ISSUE TITLE
Update JQuery version

### ISSUE THREAD

**Issue by jmichaud-ips** [okta/okta-signin-widget#378]
Can this be updated to use JQuery version 3.2.1?

**Comment by jmelberg-okta** [comment]
Hey @jmichaud-ips!

Our team has prioritized updating the versions of a few packages - including `jquery` and `jquery.cookie`. Although we cannot commit to a timeline just yet - I'm going to label this issue as an "Enhancement".

If you'd like to get a jump start - please feel free to submit a pull request and our team would be happy to take a look.

In the meantime, we'll do our best to keep this issue updated with the current status on upgrading these dependencies.

👍

**Comment by robertjd** [comment]
Hi @jmichaud-ips , can you let us know if our current jquery dependency is creating a blocker for you?  If so can you tell us a bit more about the issue to help us triage? Thanks!

**Comment by jmichaud-ips** [comment]
We have an Angular 5 website built with Angular-cli. We have multiple components that depend on jquery, different versions of course. We were able to work around the problem by NOT including the jquery package in our package.json file. And then we added "../node_modules/jquery/dist/jquery.min.js" to the .angular-cli.json file scripts array. Things are magically working for now, but jquery 1.x and 2.x are end of life and officially unsupported. Perhaps if there is need to support IE 8 or older still, a separate package could be created that depends on an older jquery version.

**Comment by nbarbettini** [comment]
Hey @jmichaud-ips, sorry to bug you again. I want to make sure I understand what you ran into. It sounds like:

- You asked for a >3.0 version of jQuery in your `package.json`
- Due to also having `okta-signin-widget` in your `package.json`, you didn't get that version
- You were able to work around this by not explicitly requesting jQuery in `package.json` and relying on the fact that it was implicitly included via `okta-signin-widget`

Does that describe your scenario? Or is there something else going on that I'm missing?

**Comment by jmichaud-ips** [comment]
Correct. The part that feels weird to us is in the .angular-cli.json, we still have to include the jquery.min.js in the scripts array, but our package.json has no jquery package listed. Maybe that is fine, I don’t know.

**Comment by rmeyer-taos** [comment]
Any update on this issue. The CVE is causing client deployment issues due to security concerns.

**Comment by ruhkopf** [comment]
Is it possible to give this issue a higher priority? `npm audit` fails due to https://nodesecurity.io/advisories/796. Perhaps a label of `vulnerability` or `bug` would be more fitting?

**Comment by swiftone** [comment]
@ruhkopf - Acknowledged.

**Comment by ayodeleopetumo** [comment]
<img width="751" alt="screenshot_2019-05-10_at_12 54 36_pm" src="https://user-images.githubusercontent.com/13363563/57629491-c824ad80-7593-11e9-8072-0b4db99a618b.png">

Any update on this please? 👆🏽

**Comment by cniesen** [comment]
FYI, the CVE-2015-9251 and CVE-2019-11358 aren't just limited to the okta-signin-widget.  The default Okta signin page also uses the old jQuery version.

I asked the question if that's a concert https://support.okta.com/help/s/question/0D51Y00006G4hhwSAB  .

**Comment by mraible** [comment]
Version 2.19.0 of the Sign In Widget has an npm vulnerability reported by `npm audit`.

```
──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Moderate      │ Prototype Pollution                                          │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ jquery                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.4.0                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ @okta/okta-signin-widget                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ @okta/okta-signin-widget > jquery                            │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/796                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
```

**Comment by Bwvolleyball** [comment]
> Version 2.19.0 of the Sign In Widget has an npm vulnerability reported by `npm audit`.

This should be the main reason this is now prioritized 😃

**Comment by swiftone** [comment]
I know all the teams involved have been watching this and related CVEs, so they're taking care of any issue on the Okta signin page, but for anyone using the widget locally: the latest version passes `npm audit` without complaint.

**Comment by aarongranick-okta** [comment]
I have confirmed this issue was fixed in the `3.0` release of the widget. Anyone experiencing this issue with a `2.x` version should upgrade to the latest version of the widget. See the [Migration Guide](https://github.com/okta/okta-signin-widget/blob/master/MIGRATING.md) for more details.
